.github/workflows/publishing.yml

```diff
@@ -742,3 +747,22 @@ jobs:
           done
         env:
           GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
+
+      - name: Delete artifacts
```

Comment:
This action only deletes the artifact named downstream_assets_${{ github.run_id }} (defined by ${{ needs.assets.outputs.retrieved_assets }}). Hence it will only ever delete the assets that are superseded by the ones created during publishing and it will only delete them if they were created during this workflow.